docs/proposal/layer4_authorization.md

```diff
@@ -56,5 +54,39 @@ struct {
    - 3.2: If the matched record shows `value=deny`, it alters the message flag, sends an RST message to the server, clears the corresponding `auth_map` record. If no record is matched, implying authorization is allowed, the message is passed through.
 4. **client retry**: The client attempts to send another message, but because the server has closed the connection, the client receives a "reset by peer" signal and subsequently closes its own channel.
 
+### Xdp authentication
+
+#### Design detail
+
+![l4_authz_xdp](pics/kmesh_l4_authorization_xdp.svg#pic_center)
+
+#### Map definition
+
+map_of_wl_policy: records the policies that are configured for the workload.
+
+map_of_authz: records the authz rules of policies.
+
+```.c
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(wl_policies_v));
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+    __uint(max_entries, MAP_SIZE_OF_AUTH_POLICY);
+} map_of_wl_policy SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(Istio__Security__Authorization));
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+    __uint(max_entries, MAP_SIZE_OF_AUTH_POLICY);
+} map_of_authz SEC(".maps");
+```
+
+#### Processing logic
 
+![l4_authz_xdp](pics/kmesh_l4_authorization_match_chain.svg#pic_center)
 
+1. Message Parsing: When a packet enters the XDP processing logic on the server side, the tuple information of the packet is parsed. The corresponding workload instance is then located based on the destination IP, and the authentication rules configured on that workload are retrieved.
+2. Rule Matching: As shown in the figure, XDP implements a matching chain logic. First, it determines whether to allow or deny the packet based on the port  info, if the result is deny, the packet is intercepted, and the process ends. If the result is allow, the next matching logic (e.g., IP matching) is called using a BPF tail call. This process is repeated until the last link in the chain. If the final result is allow, XDP\_PASS is returned, and the packet is forwarded through the kernel network stack to the server.
```

Comment:
As i talked with @weli-l the tailcall mechanism seems not feasible, because a workload may have several authz policies set, tailcall does not return.